lib/rucio/tests/test_bin_rucio.py

```diff
@@ -210,7 +210,7 @@ def test_upload_file(self):
         remove(tmp_file1)
         remove(tmp_file2)
         remove(tmp_file3)
-        nose.tools.assert_not_equal(re.search("File {0}:{1} successfully uploaded on the storage".format(self.user, tmp_file1[5:]), out), None)
+        nose.tools.assert_true("File %s:%s successfully uploaded on the storage" % (self.user, tmp_file1[5:]) in out)
```

Comment:
Any reason to change format by %?
Also, I think the test is failing because tmp_file1[5:] is bytes, not string. Maybe the solution is to cast explicitly str(tmp_file1[5:]).

Reply:
for some reason `.format()` didn't work. 
For fixing the errors, I'll suggest to do it iteratively with more Pull Requests.

Reply:
So we can merge this and fix the rest iteratively?

Reply:
yes!

Reply:
@jwackito I think ´tmp_file1[5:].decode('utf-8')´ is better

Reply:
... but maybe ´file_generator()´ should be fixed instead